typings/index.d.ts

```diff
@@ -533,7 +533,8 @@ declare module 'discord.js' {
 		public edit(data: GuildEditData, reason?: string): Promise<Guild>;
 		public equals(guild: Guild): boolean;
 		public fetchAuditLogs(options?: GuildAuditLogsFetchOptions): Promise<GuildAuditLogs>;
-		public fetchBans(): Promise<Collection<Snowflake, User>>;
+		public fetchBan(user: UserResolvable): Promise<BanInfo>;
+		public fetchBans(withReasons?: boolean): Promise<Collection<Snowflake, User | BanInfo>>;
```

Comment:
Adding overloads would be nice to not have to type assert around or break compilation for people updating.
```suggestion
		public fetchBans(withReasons?: false): Promise<Collection<Snowflake, User>>;
		public fetchBans(withReasons: true): Promise<Collection<Snowflake, BanInfo>>;
		public fetchBans(withReasons: boolean): Promise<Collection<Snowflake, BanInfo | User>>;
```